CMakeLists.txt

```
@@ -74,26 +91,26 @@ else()
    message("-- Not using MPI")
 endif()
 
-# This library
-# -----------------------------
+#[==============================================================================================[
+#                                   Core Library                                                #
+#]==============================================================================================]
 option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
 
 # Define libraries
-add_library(lib_kmeans "")
-add_library(fortuno_wrapper OBJECT "")
+add_library(lib_kmeans)
+add_library(fortuno_wrapper OBJECT)
 
 # Location of source - added after defining library variables,
 # as they're referenced in the source-level CMakeLists.txt
 add_subdirectory(src)
 
-set_target_properties(lib_kmeans
-        PROPERTIES
-        VERSION ${libkmeans_VERSION} 
-        SOVERSION ${libkmeans_VERSION_MAJOR}  
-        LIBRARY_OUTPUT_NAME kmeans
-        ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}
-        LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}
-        )
+set_target_properties(lib_kmeans PROPERTIES
+    VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}"
```

```suggestion
    VERSION "${PROJECT_VERSION}"
```
Why write is simple am I right?

---

😅 I wasn't focused on this - yes, clearly simplification now `VERSION` is defined for the project